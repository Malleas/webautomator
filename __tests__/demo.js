const {WebAutomator} = require('../index');

const automator = new WebAutomator(webDriver);

beforeEach(async function () {
  await webDriver.navigate().to(`${__dirname}/demo.html`);
});

test('Demo', async function () {
  await automator.set('First Name').to('Joyce');
  await automator.set('Last Name').to('Byers');
  await automator.set('Gender').to('Female');

  await automator.after('Child Information', async context => {
    await context.set('First Name').to('Will');
    await context.set('Last Name').to('Byers');
    await context.set('Gender').to('Male');
    await context.set('Date of Birth').to('3/22/1971');
  });

  await automator.set('Shipping Information').to('2-Day');

  await automator.click('I confirm that these details are accurate.');

  expect(await automator.getContent()).toMatchSnapshot();
});