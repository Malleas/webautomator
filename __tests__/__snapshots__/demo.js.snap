// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo 1`] = `
<window>
  Parent Information
  First Name:
  <textbox
    value="Joyce"
  />
  Last Name:
  <textbox
    value="Byers"
  />
  Gender:
  <combobox>
    <option>
      Male
    </option>
    <option
      selected="true"
    >
      Female
    </option>
  </combobox>
  Child Information
  First Name:
  <textbox
    value="Will"
  />
  Last Name:
  <textbox
    value="Byers"
  />
  Gender:
  <combobox>
    <option
      selected="true"
    >
      Male
    </option>
    <option>
      Female
    </option>
  </combobox>
  Date of Birth:
  <textbox
    value="1971-03-22"
  />
  Shipping Information
  <radio />
  Ground
  <radio
    checked="true"
  />
  2-Day
  <radio />
  1-Day
  Shipping Options
  <checkbox />
  Gift wrapped?
  <checkbox />
  Warranty?
  <checkbox
    checked="true"
  />
  Returnable?
  Confirmation
  <checkbox
    checked="true"
  />
  I confirm that these details are accurate.
</window>
`;
